internal/controller/controllers/agent_controller.go

```diff
@@ -134,6 +134,46 @@ func (r *AgentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	return result, nil
 }
 
+func (r *AgentReconciler) updateInstallerArgs(ctx context.Context, c *common.Cluster, host *common.Host, agent *aiv1beta1.Agent) error {
+
+	if agent.Spec.InstallerArgs == host.InstallerArgs {
+		r.Log.Debugf("Nothing to update, installer args were already set")
+		return nil
+	}
+
+	// InstallerArgs are saved in DB as string after unmarshalling of []string
+	// that operation removes all whitespaces between words
+	// in order to be able to validate that field didn't changed
+	// doing reverse operation
+	// If agent.Spec.InstallerArgs was not set but host.InstallerArgs was, we need to delete InstallerArgs
+	args := models.InstallerArgsParams{Args: []string{}}
```

Comment:
Consider
```suggestion
	agentSpecInstallerArgs := models.InstallerArgsParams{Args: []string{}}
```
args.Args looks funny and doesn't say much about what it is later when you compare it to host.InstallerArgs